tests/utils/test_bits.py

```diff
@@ -0,0 +1,107 @@
+from math import isinf, isnan
+from random import randrange
+
+import pytest
+
+from xdsl.utils.bits import (
+    as_str,
+    from_f16,
+    from_f32,
+    from_f64,
+    from_i8,
+    from_i16,
+    from_i32,
+    from_i64,
+    from_u8,
+    from_u16,
+    from_u32,
+    from_u64,
+    to_f16,
+    to_f32,
+    to_f64,
+    to_i8,
+    to_i16,
+    to_i32,
+    to_i64,
+    to_u8,
+    to_u16,
+    to_u32,
+    to_u64,
+)
+
+
+@pytest.mark.parametrize(
+    "i",
+    (randrange(1 << 8) for _ in range(256)),
+)
```

Comment:
Notice it is a generator instead of tuple or list. Also it is mimicking QuickCheck in Haskell. 256 is just arbitrary and consistent for all bits (8 / 16/ 32 / 64). I could have 256 for 8bits and 65536 for 16/32/64 bits.